2 Niagara Fundamentals Exercise - Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplexer.java

```diff
@@ -0,0 +1,257 @@
+   * @see #getLn2
+   * @see #setLn2
+   */
+  public static final Property ln2 = newProperty(Flags.SUMMARY, BInteger.make(2).as(BInteger.class).getInt(), null);
+
+  /**
+   * Get the {@code ln2} property.
+   * @see #ln2
+   */
+  public int getLn2() { return getInt(ln2); }
+
+  /**
+   * Set the {@code ln2} property.
+   * @see #ln2
+   */
+  public void setLn2(int v) { setInt(ln2, v, null); }
+
+  //endregion Property "ln2"
+
+  //region Property "ln3"
+
+  /**
+   * Slot for the {@code ln3} property.
+   * @see #getLn3
+   * @see #setLn3
+   */
+  public static final Property ln3 = newProperty(Flags.SUMMARY, BInteger.make(3).as(BInteger.class).getInt(), null);
+
+  /**
+   * Get the {@code ln3} property.
+   * @see #ln3
+   */
+  public int getLn3() { return getInt(ln3); }
+
+  /**
+   * Set the {@code ln3} property.
+   * @see #ln3
+   */
+  public void setLn3(int v) { setInt(ln3, v, null); }
+
+  //endregion Property "ln3"
+
+  //region Property "s0"
+
+  /**
+   * Slot for the {@code s0} property.
+   * @see #getS0
+   * @see #setS0
+   */
+  public static final Property s0 = newProperty(Flags.SUMMARY, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code s0} property.
+   * @see #s0
+   */
+  public boolean getS0() { return getBoolean(s0); }
+
+  /**
+   * Set the {@code s0} property.
+   * @see #s0
+   */
+  public void setS0(boolean v) { setBoolean(s0, v, null); }
+
+  //endregion Property "s0"
+
+  //region Property "s1"
+
+  /**
+   * Slot for the {@code s1} property.
+   * @see #getS1
+   * @see #setS1
+   */
+  public static final Property s1 = newProperty(Flags.SUMMARY, BBoolean.FALSE.as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code s1} property.
+   * @see #s1
+   */
+  public boolean getS1() { return getBoolean(s1); }
+
+  /**
+   * Set the {@code s1} property.
+   * @see #s1
+   */
+  public void setS1(boolean v) { setBoolean(s1, v, null); }
+
+  //endregion Property "s1"
+
+  //region Property "out"
+
+  /**
+   * Slot for the {@code out} property.
+   * @see #getOut
+   * @see #setOut
+   */
+  public static final Property out = newProperty(Flags.READONLY, BInteger.make(0).as(BInteger.class).getInt(), null);
+
+  /**
+   * Get the {@code out} property.
+   * @see #out
+   */
+  public int getOut() { return getInt(out); }
+
+  /**
+   * Set the {@code out} property.
+   * @see #out
+   */
+  public void setOut(int v) { setInt(out, v, null); }
+
+  //endregion Property "out"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BMultiplexer.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+  @Override
+  public void changed(Property property, Context context) {
+    super.changed(property, context);
+    //System.out.println("Property changed: " + property.getName());
+    updateOutput();
+  }
+
+  public void updateOutput () {
```

Comment:
In general, if you're going to delegate the work of your `changed` method to a separate method like this, I would probably make this method `private` so that other classes can't call `updateOutput()` directly.